src/main/java/com/thealgorithms/datastructures/graphs/BoruvkaAlgorithm.java

```diff
@@ -0,0 +1,191 @@
+package com.thealgorithms.datastructures.graphs;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Boruvka's algorithm to find Minimum Spanning Tree
+ * (https://en.wikipedia.org/wiki/Bor%C5%AFvka%27s_algorithm)
+ *
+ * @author itakurah (https://github.com/itakurah)
+ */
+
+class BoruvkaAlgorithm {
+    /**
+     * Represents an edge in the graph
+     */
+    static class Edge implements Comparable<Edge> {
+        int src, dest, weight;
+
+        public Edge(int src, int dest, int weight) {
+            this.src = src;
+            this.dest = dest;
+            this.weight = weight;
+        }
+
+        @Override
+        public int compareTo(Edge other) {
+            return Integer.compare(this.weight, other.weight);
+        }
+    }
+
+    /**
+     * Represents the graph
+     */
+    static class Graph {
+        int V, E;
```

Comment:
```suggestion
        final int V;
        final int E;
```
Additionally I would suggest to rename these members.